patches/components-update_client-updater_state_win.cc.patch

```diff
@@ -1,35 +1,32 @@
 diff --git a/components/update_client/updater_state_win.cc b/components/update_client/updater_state_win.cc
-index 89c2c186f9c095d6c2a9210ca2c4e8a94ee64b98..9f7ba1eb42bb50d21cdc2a905dd45e5d5bf7d82c 100644
+index 89c2c186f9c095d6c2a9210ca2c4e8a94ee64b98..7aa7afec5dc013785973a5ef2d13bf8367ebe5d6 100644
 --- a/components/update_client/updater_state_win.cc
 +++ b/components/update_client/updater_state_win.cc
-@@ -25,20 +25,35 @@ namespace {
+@@ -25,20 +25,32 @@ namespace {
  
  // Google Update group policy settings.
  const wchar_t kGoogleUpdatePoliciesKey[] =
-+#if defined(BRAVE_CHROMIUM_BUILD)
 +    L"SOFTWARE\\Policies\\BraveSoftware\\Update";
-+#else
++#if 0
      L"SOFTWARE\\Policies\\Google\\Update";
 +#endif
  const wchar_t kCheckPeriodOverrideMinutes[] = L"AutoUpdateCheckPeriodMinutes";
  const wchar_t kUpdatePolicyValue[] = L"UpdateDefault";
  const wchar_t kChromeUpdatePolicyOverride[] =
-+#if defined(BRAVE_CHROMIUM_BUILD)
 +    L"Update{AFE6A462-C574-4B8A-AF43-4CC60DF4563B}";
-+#else
++#if 0
      L"Update{8A69D345-D564-463C-AFF1-A69D9E530F96}";
 +#endif
  
  // Don't allow update periods longer than six weeks (Chrome release cadence).
  const int kCheckPeriodOverrideMinutesMax = 60 * 24 * 7 * 6;
  
  // Google Update registry settings.
-+#if defined(BRAVE_CHROMIUM_BUILD)
 +const wchar_t kRegPathGoogleUpdate[] = L"Software\\BraveSoftware\\Update";
```

Comment:
for future work: all of these potentitally can be redefined via chromium_src?

Reply:
Yes, I had the same thought exactly, but for this patch I think it's probably better to keep it like this so that it's more clear what the change is about?

Reply:
@iefremov I've been looking into this today in the context of other similar situations where I could actually remove the definitions from the .patch file and put them into an override (i.e. https://github.com/brave/brave-core/pull/8063, https://github.com/brave/brave-core/pull/8061 and https://github.com/brave/brave-core/pull/8062), but in this particular case I'm not sure it's a good idea.

My reasoning is that in this case using a chromium_src override to define Brave-specific values for these variables won't work as expected, since we'd need first to redefine the ones from upstream into something like `<variable_name>_ChromiumImpl` or `<variable_name>_Unused` (depending on whether we will actually use them -e.g. for the `!defined(BRAVE_CHROMIUM_BUILD)` case- or not), but doing that will also change the places where such variables are already being referenced inside this `updater_state_win.cc` file.

If this file didn't contain any logic referencing those variables and only contained definitions (as it's the case for the PRs linked above), I'd say this approach would work. But considering that's not the case and they are indeed being referenced, I'm not sure how to do this in a way that is actually better / less bad than using a .patch file.

WDYT? Do you have any suggestion?

Reply:
no suggestions, let's leave them alone :)